docs/pages/repo/docs/core-concepts/caching.mdx

```diff
@@ -83,6 +83,85 @@ turbo run build --force
 
 Note that `--force` disables cache reads but does not disable cache writes. If you want to disable cache writes, use the `--no-cache` flag.
 
+## Handling Node.js versions
+
+To account for Node.js versions, use [the `engines` key in package.json](https://docs.npmjs.com/cli/v10/configuring-npm/package-json#engines). Turborepo will see the changes to your `package.json` and miss cache when the field is updated.
+
+## Handling platforms and other arbitrary hash contributors
+
+For advanced use cases, you may need to account for operating systems (OS), architecture, or other external factors that you want to contribute to your hash. You can do this in four quick steps.
+
+### 1. Write an arbitrary file to disk
+
+First, create a script that accounts for the hash contributors that you are interested in. For example, here is a Node.js script that looks at platform and architecture and writes those details to a file (`turbo-cache-key.json`):
+
+```js
+#!/usr/bin/env node
+
+const { writeFileSync } = require('fs');
+const { join } = require('path');
+
+const { platform, arch } = process;
+const file = join(__dirname, '..', 'turbo-cache-key.json');
```

Comment:
We should explain this, what directory _should_ it be in, package root? monorepo root?

Reply:
Too perscriptive. Fixed it for the opposite, just naming a file.